crates/sui-core/src/transaction_input_checker.rs

```diff
@@ -96,12 +92,8 @@ pub(crate) async fn check_dev_inspect_input(
         if !object.is_immutable() {
             fp_ensure!(
                 used_objects.insert(object.id().into()),
-                SuiError::InvalidBatchTransaction {
-                    error: format!(
-                        "Mutable object {} cannot appear in more than one single \
-                        transactions in a batch",
-                        object.id()
-                    ),
+                UserInputError::MutableObjectUsedMoreThanOnce {
```

Comment:
Mostly noting for myself, we will need to make sure that it is entirely unique for programmable transactions